libzecale/core/aggregator_circuit_wrapper.hpp

```diff
@@ -36,7 +36,7 @@ class aggregator_circuit_wrapper
     /// Generate a proof and returns an extended proof
     extended_proof<wppT, wSnarkT> prove(
         typename nSnarkT::VerificationKeyT nested_vk,
-        std::array<libzeth::extended_proof<nppT, nSnarkT>, NumProofs>
+        std::array<const libzeth::extended_proof<nppT, nSnarkT> *, NumProofs>
```

Comment:
What about passing arguments by reference here?

Reply:
Changed to const ref.